### ISSUE TITLE
Auto-On when unlocked

### ISSUE THREAD

**Issue by tophee** [raine/Lolgato#1]
Thanks for the app!

I have a suggestion: it would be great to have an option to turn the lights _on_ automatically when the computer is unlocked/woken from sleep, but only if it was previously turned off automatically. In other words, it would probably be a nuisance to always have the lights turn on when you wake/unlock you Mac, but if the lights were turned off automatically (because the Mac went to sleep/was locked, then it makes sense to turn them back on when the Mac is unlocvked again. - Although, perhaps adding another condition might improve this feature further: if the most recent auto-off was on the same date.

**Comment by raine** [comment]
Hey, thanks for the suggestion.

At the time when I did the auto-off on lock/sleep, I considered this but figured that user's context might've changed during the time away so that turning on the lights automatically on unlock might not be what the user wants always. But that was mostly a guess, not based on real usage. If in your personal experience this would be a useful addition then I think it makes sense to consider adding it.

**Comment by tophee** [comment]
Yes, It would definitely have to be an option as it depends on how you use the lights. I use them a lot as my desk light, even when not using the camera and for that, it would be very useful if the lights went on when I open the computer and off when I close it. For users who only use the lights on camera, this is obviously not useful.

A related idea would be to provide an option to adjust the temperature based on time of day/ hours until sunset. Before I bought the keylight, I somehow assumed that all of this kind of automation was already part of the standard functionality. In retrospect, I'm not sure why I thought so. Probably because they're so expensive, so I just assumed that they would have obviously useful and easy to implement automations. So, I'm glad that you're sharing this app :-)

**Comment by raine** [comment]
> A related idea would be to provide an option to adjust the temperature based on time of day/ hours until sunset.

That's a really cool idea.

> A related idea would be to provide an option to adjust the temperature based on time of day/ hours until sunset. Before I bought the keylight, I somehow assumed that all of this kind of automation was already part of the standard functionality. In retrospect, I'm not sure why I thought so. Probably because they're so expensive, so I just assumed that they would have obviously useful and easy to implement automations. So, I'm glad that you're sharing this app :-)

Hehe, I'm glad you're finding it useful. I bought the light with the "hackability" in mind and decided already before buying the light that I will make this app if these features are not available in the official app.

**Comment by raine** [comment]
I've made the sleep option now work so that it will also turn on lights if they were turned off by it on sleep or lock.

<img width="395" alt="CleanShot 2024-10-31 at 23 47 03@2x 1" src="https://github.com/user-attachments/assets/d7b02ced-98a0-418a-a07b-85f4091a0773">

**Comment by tophee** [comment]
Great! I will try it out soon. 

Quick comment: For me, I think it works like this, with a single setting, but I think for some people it might be better to have separate settings for auto-on and auto-off.
